### PR TITLE
Fix TTS adapter tutorial

### PR DESCRIPTION
**Pull request by hsiehjackson**
# What does this PR do ?

Fix adapter tutorial to compatible with latest pre-traiend FastPitch checkpoint

**Collection**: [TTS]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/tts/modules/submodules.py [hunk 1]
```diff
@@ -758,15 +758,11 @@ def forward(self, batch_size=None, speaker=None, reference_spec=None, reference_
...
-            if reference_spec is None or reference_spec_lens is None:
-                raise ValueError(
-                    "You should add `reference_audio` in sup_data_types or remove `speaker_encoder`in config."
-                )
-            out = self.gst_module(reference_spec, reference_spec_lens)
-            embs = out if embs is None else embs + out
-
-        elif self.gst_module is None and reference_spec is not None and reference_spec_lens is not None:
-            logging.warning("You may add `gst_module` in speaker_encoder to use reference_audio.")
+        if reference_spec is not None and reference_spec_lens is not None:
```
Comment (subhankar-ghosh) [inline nemo/collections/tts/modules/submodules.py:761]: why was the check on self.gst_module removed?

File: tutorials/tts/FastPitch_Adapter_Finetuning.ipynb [hunk 2]
```diff
@@ -645,17 +647,7 @@
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "0908431a",
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "wave_model = WaveformFeaturizer(sample_rate=sample_rate)"
```
Comment (subhankar-ghosh) [inline tutorials/tts/FastPitch_Adapter_Finetuning.ipynb:652]: why was this removed? This object is needed for converting wav -> features

Comment (hsiehjackson) [reply tutorials/tts/FastPitch_Adapter_Finetuning.ipynb:652]: We don't need to use `reference_spec` during inference, so we don't need to get the audio features. The only thing we need is the model and text.

**Review by subhankar-ghosh — APPROVED** [review]
LGTM.
